src/ray/raylet/scheduling/cluster_task_manager_test.cc

```diff
@@ -284,8 +288,8 @@ TEST_F(ClusterTaskManagerTest, ResourceTakenWhileResolving) {
       std::make_shared<MockWorker>(WorkerID::FromRandom(), 1234);
   std::shared_ptr<MockWorker> worker2 =
       std::make_shared<MockWorker>(WorkerID::FromRandom(), 12345);
-  pool_.PushWorker(std::dynamic_pointer_cast<WorkerInterface>(worker2));
-  pool_.PushWorker(std::dynamic_pointer_cast<WorkerInterface>(worker));
+  pool_.PushWorker(std::static_pointer_cast<WorkerInterface>(worker2));
```

Comment:
It's irrelevant with the PR and hope it's not annoying. 
`static_cast` here is enough and I feel it's not worth putting it in another PR.